src/net/mod.rs

```diff
@@ -159,8 +160,13 @@ where
             UpdateState::Updated => NetworkState::Updated,
         };
 
-        match self.miniconf.update() {
-            Ok(true) => NetworkState::SettingsChanged,
+        let mut settings_path = String::new();
+        match self.miniconf.handled_update(|path, old, new| {
+            settings_path.push_str(path).unwrap();
+            *old = new.clone();
```

Comment:
There's no need to perform this clone - miniconf does it internally. The only reason `old` and `new` are provided is so the user can detect what the value transition was on the given path.

Reply:
Ok, what's the reason for it being performed here?
https://github.com/quartiq/miniconf/blob/98438ab27794fbe0ad9b43a556b87241ad4097e8/src/mqtt_client/mod.rs#L433

Reply:
Ah you're correct - I'm mistaken here, this is required currently.

I don't see this mentioned in the `miniconf` docs anywhere, so I'll open an issue to clarify this. Thanks for pointing this out to me!

Reply:
I've opened https://github.com/quartiq/miniconf/issues/93 to clarify this. I think we could easily incorporate this into miniconf, but as-is, this implementation is correct.

Reply:
Ha, I'm happy that I understood what this closure does here, I was already disappointed.. ;)